pkg/synchronization/controller.go

```diff
@@ -534,6 +539,38 @@ func (c *Controller) createOrUpdateSecret(logger *zap.Logger, secretSpec *core_v
 	return true, secret, err
 }
 
+func (c *Controller) ensureCommonSecretExists(logger *zap.Logger, ns *core_v1.Namespace) (bool /* retriable */, error) {
+	_, err := c.MainClient.CoreV1().Secrets(ns.Name).Get(apik8scompute.CommonSecretName, meta_v1.GetOptions{})
+
+	// nothing to do, it exists; nothing to update
+	if err == nil {
```

Comment:
I think it might be better to use `switch` here to show that there are a few distinct cases to handle. WDYT?

Reply:
I agree that might be slightly cleaner. Is there a good way to structure that with `api_errors.IsNotFound` though?

Reply:
https://github.com/golang/go/wiki/Switch#missing-expression

Reply:
Updated